src/wrapper_array.jl

```diff
@@ -166,14 +166,22 @@ for f in (:getindex, :view, :dotview)
         end
 
         @propagate_inbounds function Base.$f(a::NamedDimsArray, inds...)
-            # Some nonscalar case, will return an array, so need to give that names.
+            # Some nonscalar case, will return an array, probably with names
             data = Base.$f(parent(a), inds...)
             L = remaining_dimnames_from_indexing(dimnames(a), inds)
-            return NamedDimsArray{L}(data)
+            if L === nothing
+                # All names dropped, e.g. from mat[mat .> 0]
```

Comment:
Why though?
I assume it is because there are no valid names for this because it has different shape to the inputs?
Comment needs to explain

Reply:
What should it say? The output dimension isn't one of the input dimensions, as in the issue... it's not even a meaningful combination of them like `vec(x)` would be. Perhaps you have a sentence in mind?

Reply:
I actually don't know what its doing at all, is why I want the comment

Reply:
Oh after thinking a bit i see there is no other way for it to be.
Maybe we can say something like: 

> `mat[mat .> 0.5] is the same as `vec(mat)[vec(mat .> 0.5)]` and so ravels the n dimension down to one new combined one.

Reply:
ok, added that a few lines down